security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysis.java

```diff
@@ -89,12 +111,85 @@ protected SecurityAnalysisResult readResults(Path workingDir) {
          */
         @Override
         protected List<CommandExecution> buildCommandExecution() {
-            SimpleCommand cmd = baseCommand("security-analysis-task")
+            return Collections.singletonList(new CommandExecution(baseCmdBuilder().build(), actualTaskCount, 1));
+        }
+
+        SimpleCommandBuilder baseCmdBuilder() {
+            return baseCommand(SA_TASK_CMD_ID)
                     .option(TASK, i -> new Partition(i + 1, actualTaskCount).toString())
-                    .option("output-file", this::getOutputFileName)
-                    .option("output-format", "JSON")
+                    .option(OUTPUT_FILE_OPTION, this::getOutputFileName)
+                    .option(OUTPUT_FORMAT_OPTION, "JSON");
+        }
+
+    }
+
+    private class SubTaskWithLogHandler extends AbstractExecutionHandler<SecurityAnalysisResultWithLog> {
+
+        private SubTaskHandler simpleSubTaskHandler;
+
+        private static final String SA_LOG_TAR = "sa_logs.tar";
+        private static final String SA_LOG_TAR_GZ = SA_LOG_TAR + ".gz";
+
+        public SubTaskWithLogHandler(SubTaskHandler subTaskHandler) {
+            this.simpleSubTaskHandler = Objects.requireNonNull(subTaskHandler);
+        }
+
+        @Override
+        public List<CommandExecution> before(Path workingDir) throws IOException {
+            simpleSubTaskHandler.before(workingDir); // invocation just for prepare input files
+            // override cmd with log
+            SimpleCommand cmd = simpleSubTaskHandler.baseCmdBuilder()
+                    .option(OUTPUT_LOG_OPTION, this::getLogOutputName)
                     .build();
-            return Collections.singletonList(new CommandExecution(cmd, actualTaskCount, 1));
+            return Collections.singletonList(new CommandExecution(cmd, simpleSubTaskHandler.actualTaskCount, 1));
+        }
+
+        private String getLogOutputName(int taskNumber) {
+            return "logs_" + taskNumber + ".xml.gz";
+        }
+
+        @Override
+        public SecurityAnalysisResultWithLog after(Path workingDir, ExecutionReport report) throws IOException {
+            SecurityAnalysisResult re = simpleSubTaskHandler.after(workingDir, report);
+            archiveLogs(workingDir);
+            compressLogs(workingDir);
+            Files.delete(workingDir.resolve(SA_LOG_TAR));
+            return new SecurityAnalysisResultWithLog(re, workingDir.resolve(SA_LOG_TAR_GZ).toFile());
+        }
+
+        private void archiveLogs(Path workingDir) throws IOException {
+            try (TarArchiveOutputStream taos = new TarArchiveOutputStream(new FileOutputStream(workingDir.resolve(SA_LOG_TAR).toFile()))) {
```

Comment:
Shouldn't we work with an in memory stream, instead of adding that file in the command execution result directory ? Here we create IO to filesystem which I think is not necessary.